prow/flagutil/storage.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/pkg/io"
+)
+
+type StorageClientOptions struct {
+	// The following are used for reading/writing to GCS.
+	GCSCredentialsFile   string
+	GCSOptionDescription string
```

Comment:
nit: why is this public, or stored? couldn't we store a closure to bind onto a flagset or at least store the description as a private member?

Reply:
Thank you for the review. Fixed.